vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -1,16 +1,39 @@
 ---
-  - name: Get the name of a worker node
+  - name: Get the name of a worker bmh name
     shell: |
        kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
        |select(.status.provisioning.state | contains("provisioned")) |select(.spec.consumerRef.name
        | contains("{{ CLUSTER_NAME }}-workers")) | .metadata.name'
-    register: worker_node_name
+    register: worker_bmh_name
 
-  - name: Get the names of master nodes
+  - name: Get the names of master bmh name
     shell: |
        kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
        |select(.status.provisioning.state | contains("provisioned")) |select(.spec.consumerRef.name
        | contains("{{ CLUSTER_NAME }}-controlplane")) | .metadata.name'
+    register: master_bmh_name
+
+  - set_fact:
+      WORKER_BMH: "{{ worker_bmh_name.stdout }}"
+      MASTER_BMH_0: "{{ master_bmh_name.stdout_lines.0 }}"
+      MASTER_BMH_1: "{{ master_bmh_name.stdout_lines.1 }}"
+      MASTER_BMH_2: "{{ master_bmh_name.stdout_lines.2 }}"
+  
+  - set_fact:
+      WORKER_NODE_VM: "{{ WORKER_BMH | replace('-','_') }}"
+      MASTER_NODE_0_VM: "{{ MASTER_BMH_0 | replace('-','_') }}"
+      MASTER_NODE_1_VM: "{{ MASTER_BMH_1 | replace('-','_') }}"
+      MASTER_NODE_2_VM: "{{ MASTER_BMH_2 | replace('-','_') }}"
+
+  - name: Obtain the kubeconfig from target cluster
+    shell: "kubectl get secrets {{ CLUSTER_NAME }}-kubeconfig -n {{ NAMESPACE }} -o json | jq -r '.data.value'| base64 -d > /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    
+  - name: Get the name of the worker node name
+    shell: "kubectl get nodes --kubeconfig /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml | grep none | awk '{{print $1}}'"
+    register: worker_node_name
```

Comment:
do you need that ? the node name is the machine name if I recall correctly

Reply:
Yes you are right, machine name and node name are same. If we get machines info,  have only machine name and running status info and its difficult to filter the worker and master from that info. But if we get the k8s node info, its easy to filter master and worker node. That is the reason I think we need this here.

Reply:
actually is is really easy to filter master and worker machines, check the annotations ;)